src/components/PostDetails.tsx

```diff
@@ -98,21 +62,21 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
       </div>
 
       <div className="block">
-        {!loaded && <Loader />}
+        {!loading && <Loader />}
 
-        {loaded && hasError && (
+        {loading && err && (
```

Comment:
The condition here checks for `loading && err`, which means it will show an error message while loading is still true. Ensure that this is the intended behavior. Typically, errors are shown when loading has completed, i.e., `!loading && err`.